Automattic/wp-calypso
# Add react-query migrations for email forwards management

#### Proposed Changes

This is the 2nd PR (follow up to #64569) in the process of migrating from Redux actions to react-query mutations in the email forwards management section. 

Here's a summary of what this PR changes:
- Two new mutations, one for adding email forwards and one for resending the verification email to the target email address.
- The add and remove mutations optimistically mutate the data from the relevant fetch queries.
- Since we now have optimistic mutations, adding a new email forward is an instant operation. I've adapted the relevant components accordingly.

#### Testing Instructions



To fully test the changes, it's easiest to start with a domain without any email solution and then test in a specific order.

1. Have a domain without any email solution
2. Navigate to `/email/:domain/manage/:site_slug`
3. Click the "Start with Email Forwarding" link at the bottom of the page
4. Add a new email forward
5. Ensure that you are immediately taken to `/email/:domain/manage/:site_slug`
6. Ensure that you immediately see the newly added email forward on that page
7. Ensure that after a second or two, the newly added email forward displays a "Verification needed" warning
8. Click the "Resend verification email" button
9. Ensure that the destination email address receives another "Verify your email forwarding address" email
10. Remove the email forward
11. Ensure that it immediately disappears
12. Ensure that you see a notification after a second or two about a successful removal
13. Ensure that you are automatically taken to the "Pick an email solution" view after a few seconds



Related to #64569
